### PR TITLE
limit secret permissions

### PR DESCRIPTION
**Pull request by zhiweiyin318**

### CODE REVIEW

File: pkg/operators/clustermanager/options.go [hunk 1]
```diff
@@ -31,9 +35,27 @@ func (o *Options) RunClusterManagerOperator(ctx context.Context, controllerConte
 	// kubeInformer is for 3 usages: configmapInformer, secretInformer, deploynmentInformer
 	// After we introduced hosted mode, the hub components could be installed in a customized namespace.(Before that, it only inform from "open-cluster-management-hub" namespace)
 	// It requires us to add filter for each Informer respectively.
-	// TODO: Wathc all namespace may cause performance issue.
+	// TODO: Watch all namespace may cause performance issue.
 	kubeInformer := informers.NewSharedInformerFactoryWithOptions(kubeClient, 5*time.Minute)
```
Comment (qiujian16) [inline pkg/operators/clustermanager/options.go:39]: why do we still need this?

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_runtime_reconcile.go [hunk 2]
```diff
@@ -187,10 +187,10 @@ func (c *runtimeReconcile) clean(ctx context.Context, cm *operatorapiv1.ClusterM
 // getSAs return serviceaccount names of all hub components
 func getSAs(clusterManagerName string) []string {
 	return []string{
-		clusterManagerName + "-registration-controller-sa",
-		clusterManagerName + "-registration-webhook-sa",
-		clusterManagerName + "-work-webhook-sa",
-		clusterManagerName + "-placement-controller-sa",
-		clusterManagerName + "-work-controller-sa",
+		"registration-controller-sa",
```
Comment (qiujian16) [inline pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_runtime_reconcile.go:190]: will this impact upgrade?

File: pkg/operators/clustermanager/options.go [hunk 3]
```diff
@@ -79,6 +101,10 @@ func (o *Options) RunClusterManagerOperator(ctx context.Context, controllerConte
 
 	go operatorInformer.Start(ctx.Done())
 	go kubeInformer.Start(ctx.Done())
+	go singerSecretInformer.Start(ctx.Done())
```
Comment (qiujian16) [inline pkg/operators/clustermanager/options.go]: signerSecretInformer

File: pkg/operators/clustermanager/options.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/operators/clustermanager/options.go:39]: statusController and clusterManagerController use it to watch deployments

File: pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_runtime_reconcile.go (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_runtime_reconcile.go:190]: only used in hosted mode. and the impact is the old sa kubeconfig secrets will be left over.

File: pkg/operators/clustermanager/options.go (hunk 3 above)
Comment (zhiweiyin318) [reply pkg/operators/clustermanager/options.go]: fixed.

**Comment by xuezhaojun** [comment]
/hold

hosted mode part looks good to me, unhold when you think the rest parts are ready.

**Pull request merged**